demo-gallery/src/main/java/io/crossbar/autobahn/demogallery/netty/EchoClient.java

```diff
@@ -92,30 +100,46 @@ public void onJoinHandler(SessionDetails details) {
             return null;
         });
 
+        // Here we subscribe to a remote procedure.
         CompletableFuture<Subscription> counterRes = mSession.subscribe(
                 "com.example.oncounter", this::onCounter, null);
 
         counterRes.thenAccept(subscription -> System.out.println("subscribed to topic: " + subscription.topic));
 
+        // Here we register a remote procedure.
+        CompletableFuture<Registration> regFuture = mSession.register(
+                "com.example.add", this::add2, null);
+        regFuture.thenAccept(registration -> System.out.println("Registered procedure: " + registration.procedure));
+
+        // Here we publish an event.
         PublishOptions options = new PublishOptions(true, true);
         List<Object> argsCounter = new ArrayList<>();
         argsCounter.add(details.sessionID);
         argsCounter.add("Java");
-        int i = 1;
-        while (true) {
-            argsCounter.add(0, i);
+        final int[] i = new int[1];
+        ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+        executorService.scheduleAtFixedRate(() -> {
+            argsCounter.add(0, i[0]);
             CompletableFuture<Publication> pubFuture = mSession.publish(
                     "com.example.oncounter", argsCounter, null, options);
             pubFuture.thenAccept(publication -> System.out.println("published: " + publication.publication));
-            try {
-                pubFuture.get();
-                Thread.sleep(1000);
-            } catch (InterruptedException | ExecutionException e) {
-                e.printStackTrace();
-            }
             argsCounter.remove(0);
-            i += 1;
-        }
+            i[0] += 1;
+        }, 0, 1, TimeUnit.SECONDS);
+    }
+
+    private CompletableFuture<InvocationResult> add2(List<Object> args, Map<String, Object> kwargs,
+                                                     InvocationDetails details) {
+        CompletableFuture<InvocationResult> future = new CompletableFuture<>();
```

Comment:
in this basic example, we can just return a plain InvocationResult - the other side should treat it with what in Twisted is called `maybeDeferred`, and which allows uniform treatment of plain and future wrapped values.